src/tables/os2.rs

```diff
@@ -424,27 +424,57 @@ impl<'a> Table<'a> {
     #[inline]
     pub fn permissions(&self) -> Option<Permissions> {
         let n = Stream::read_at::<u16>(self.data, TYPE_OFFSET).unwrap_or(0);
-        match n & 0xF {
-            0 => Some(Permissions::Installable),
-            2 => Some(Permissions::Restricted),
-            4 => Some(Permissions::PreviewAndPrint),
-            8 => Some(Permissions::Editable),
-            _ => None,
+        if self.version <= 2 {
+            // Version 2 and prior, applications are allowed to take
+            // the most permissive of provided flags
+            let permission = if n & 0xF == 0 {
+                Permissions::Installable
+            } else if n & 8 != 0 {
+                Permissions::Editable
+            } else if n & 4 != 0 {
+                Permissions::PreviewAndPrint
+            } else {
+                Permissions::Restricted
+            };
+
+            Some(permission)
+        } else {
+            // Version 3 onwards, flags must be mutually exclusive.
+            match n & 0xF {
+                0 => Some(Permissions::Installable),
+                2 => Some(Permissions::Restricted),
+                4 => Some(Permissions::PreviewAndPrint),
+                8 => Some(Permissions::Editable),
+                _ => None,
+            }
         }
     }
 
-    /// Checks if the face subsetting is allowed.
+    /// Checks if the face allows embedding a subset, further restricted by [`Self::permissions`].
     #[inline]
     pub fn is_subsetting_allowed(&self) -> bool {
-        let n = Stream::read_at::<u16>(self.data, TYPE_OFFSET).unwrap_or(0);
-        n & 0x0100 == 0
+        if self.version <= 1 {
+            // Flag introduced in version 2
+            true
+        } else {
+            let n = Stream::read_at::<u16>(self.data, TYPE_OFFSET).unwrap_or(0);
```

Comment:
Wouldn't we always return true for old versions because of `unwrap_or(0)`? I agree that your code is more readable, just clarifying the logic.

Reply:
Old versions (specifically versions 0 and 1) neglected to specify that unused bits are zeroed, so they could have any value: "Applications must ignore bits 4 to 15 when reading a version 0 or version 1 table."
While unlikely any fonts in the wild actually did write non-zeroes here, might as well be strict with the implementation x3

Reply:
Oh right, my bad. I thought we're slicing the exact length, therefore read out of bound would always return 0.